### PR TITLE
pycord

### PR DESCRIPTION
**Pull request by classabbyamp**
- Makefile: update default python version to 3.11
- utils/resources_manager.py: use httpx instead of requests
- migrate to pycord
- update changelog, bump release

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # Those are the defaults; they can be over-ridden if specified
 # at en environment level or as 'make' arguments.
 BOTENV ?= botenv
-PYTHON_BIN ?= python3.9
+PYTHON_BIN ?= python3.11
```
Comment (0x5c) [inline Makefile:15]: equivalent change should also be done in run.sh

File: common.py [hunk 2]
```diff
@@ -161,7 +161,8 @@ async def convert(self, ctx: commands.Context, argument: str):
 def embed_factory(ctx: commands.Context) -> discord.Embed:
     """Creates an embed with neutral colour and standard footer."""
     embed = discord.Embed(timestamp=datetime.utcnow(), colour=colours.neutral)
-    embed.set_footer(text=str(ctx.author), icon_url=str(ctx.author.avatar_url))
+    if ctx.author and ctx.author.avatar:
```
Comment (0x5c) [inline common.py:164]: is this related to some new caching behaviour?

File: exts/base.py [hunk 3]
```diff
@@ -177,7 +178,7 @@ def __init__(self, bot: commands.Bot):
 
     @commands.Cog.listener()
     async def on_ready(self):
-        if not self.bot_invite:
+        if not self.bot_invite and self.bot.user:
```
Comment (0x5c) [inline exts/base.py:181]: How is is possible for the bot to have no user? QRM being a bot based on traditional text commands, this should not be possible.

File: utils/resources_manager.py [hunk 4]
```diff
@@ -28,8 +28,9 @@ def parse_index(self, index: str):
     def sync_fetch(self, filepath: str):
         """Fetches files in sync mode."""
         self.print_msg(f"Fetching {filepath}", "sync")
-        with requests.get(self.url + filepath) as resp:
-            return resp.content
+        resp = httpx.get(self.url + filepath)
```
Comment (0x5c) [inline utils/resources_manager.py:31]: I think this change leads to connections never being closed, this should be verified

File: exts/base.py [hunk 5]
```diff
@@ -141,7 +141,8 @@ async def send_group_help(self, group):
         embed.title = await self.get_command_signature(group)
         embed.description = group.help
         for cmd in await self.filter_commands(group.commands, sort=True):
-            embed.add_field(name=await self.get_command_signature(cmd), value=cmd.help, inline=False)
+            embed.add_field(name=await self.get_command_signature(cmd), value=cmd.help if cmd.help else "",
```
Comment (0x5c) [inline exts/base.py:144]: This seems odd, do they really no longer accept null values there?

File: common.py (hunk 2 above)
Comment (classabbyamp) [reply common.py:164]: it's more for mypy

File: exts/base.py (hunk 5 above)
Comment (classabbyamp) [reply exts/base.py:144]: might have been for mypy

**Review by 0x5c — APPROVED** [review]
